packages/core/core/src/Parcel.js

```diff
@@ -279,11 +279,19 @@ export default class Parcel {
       dumpGraphToGraphViz(assetGraph, 'MainAssetGraph');
 
       // $FlowFixMe Added in Flow 0.121.0 upgrade in #4381
-      let bundleGraph = await this.#bundlerRunner.bundle(assetGraph, {signal});
+      let [
+        bundleGraph,
+        serializedBundleGraph,
```

Comment:
Hmm this will be interesting once these are separate requests... I know you thought about using a WeakMap in the serializer. What was the reason for not going with that approach? Would also enable you to avoid passing the buffer around so many places in addition to the original object.